src/pages/api/user/login.ts

```diff
@@ -1,19 +1,54 @@
-// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
-import type { NextApiRequest, NextApiResponse } from 'next'
-
-export interface LoginRequestType {
-  name: string,
-}
-
-export interface LoginResponseType {
-  error?: string,
-}
-
-export default function handler(
-  req: NextApiRequest,
-  res: NextApiResponse<LoginResponseType>
-) {
-  
-
-  res.status(200).json({ error: 'John Doe' })
+import type { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from '../../../lib/prisma'
+
+// API Inputs.
+export interface LoginRequestType {
+    firstName: string,
+    lastName: string,
+    email: string,
+    password: string,
+}
+
+// API Outputs.
+export interface LoginResponseType {
+    user?: Object,
+    error: string,
+}
+
+export default async function handler (
+    req: NextApiRequest,
+    res: NextApiResponse<LoginResponseType>
+) {
+    const { email, password } = req.body;
+
+    // Error: Not all fields are filled out.
+    if(!email || !password) {
+        res.status(400);
+
+        throw new Error("Please add all fields.");
+    }
+
+    // Error: Not a valid email.
+    if((typeof email !== 'string') || (!email.includes('@')) || (!email.includes('.'))) {
+        res.status(400);
+
+        throw new Error("Please input a valid email.");
+    }
+
+    // If all checks are passed.
+
+    //Hash Password.
+
+    // Query the database.
+    const user = await prisma.user.findFirst({
+        where: {
+            email: email,
+            password: password,
+        },

```

Comment:
Since they are the same name you can write `{ email, password }`

Reply:
I'll give it a shot.